packages/cozy-clisk/src/launcher/saveFiles.js

```diff
@@ -301,13 +343,46 @@ function noMetadataDeduplicationErrors(options) {
   }
 }
 
+/**
+ * If createFile fails for non existing destination folder, create the destrination folder and
+ * retry createFile. This is done for the case where the destination folder is removed during the
+ * execution of the konnector
+ */
+async function createFileWithFolderOnError(
+  client,
+  entry,
+  options,
+  method,
+  file
+) {
+  try {
+    return await createFile(client, entry, options, method, file)
+  } catch (err) {
+    if (err.response?.status === 404) {
+      const subPath = entry.subPath || options.subPath
+      const finalPath = subPath
+        ? options.folderPath + '/' + subPath
+        : options.folderPath
+
+      const createPathResult = await client
+        .collection('io.cozy.files')
+        .createDirectoryByPath(finalPath)
+      options.folderMap.set(finalPath, createPathResult.data._id)
+      return await createFile(client, entry, options, method, file)
+    }
+    throw err
+  }
+}
+
 async function createFile(client, entry, options, method, file) {
-  const folder = await client
-    .collection('io.cozy.files')
-    .statByPath(options.folderPath)
+  const subPath = entry.subPath || options.subPath
+  const finalPath = subPath
+    ? options.folderPath + '/' + subPath
+    : options.folderPath
+  const dirId = options.folderMap.get(finalPath)
```

Comment:
We should not rely on that. Can't you pass the `dirId` to the createFile method? Since you have it from here: 

https://github.com/konnectors/libs/pull/973/commits/77234ae36ecb5188b83b97e540bea960148c938e#diff-fa7929926d4c673754905d5b4b3b9f669c0d61759edf0cc75dfd7b0a22e733f3R367-R369

Reply:
Here we can do it yes. Done